src/db/db-helper.js

```diff
@@ -241,22 +245,92 @@ module.exports = class DBHelper {
 
   static async updateEventStatusWithdrawing(currBlockTime) {
     try {
-      await db.Events.update(
+      const updatedEvents = await db.Events.update(
         {
           txStatus: TX_STATUS.SUCCESS,
           $not: { status: EVENT_STATUS.WITHDRAWING },
           arbitrationEndTime: { $lte: currBlockTime },
           currentRound: { $gt: 0 },
         },
         { $set: { status: EVENT_STATUS.WITHDRAWING } },
-        { multi: true },
+        {
+          multi: true,
+          returnUpdatedDocs: true,
+        },
       );
+      await DBHelper.updateLeaderboardWithdrawing(updatedEvents);
     } catch (err) {
       logger.error(`UPDATE Event Status Withdrawing error: ${err.message}`);
       throw err;
     }
   }
 
+  static async updateLeaderboardWithdrawing(events) {
+    try {
+      each(events, async (event) => {
+        // get all the participants
+        const bets = await DBHelper.findBet({ eventAddress: event.address });
+        const filtered = [];
+        each(bets, (bet) => {
```

Comment:
1. use lodash's reduce to accumulate all the bets in order to get the `investment`
2. keep track of the winners only. in the loop below this, you are calling `calculateWinnings` on the loser's bets as well so it's making unnecessary calls.